gfp.go

```diff
@@ -65,3 +104,21 @@ func (e *gfP) Unmarshal(in []byte) {
 
 func montEncode(c, a *gfP) { gfpMul(c, a, r2) }
 func montDecode(c, a *gfP) { gfpMul(c, a, &gfP{1}) }
+
+var one = *newGFp(1)
+
+func legendre(e *gfP) int {
+	if *e == [4]uint64{} {
+		return 0
+	}
+	// pL is (p-1)/2
+	pL := [4]uint64{0x0c2e56362f044b33, 0xf72dc468905adacf, 0xd537f65c30c26e10, 0x47da80f1a551c3fc}
+	f := &gfP{}
+	f.Exp(e, pL)
+
+	if *f == one {
```

Comment:
The first check at line 111 can be removed.

```suggestion
      montDecode(f, f)
	if *f != gfp{} {
	    return 2*int(f[0]&1)-1
	}
	return 0
}
```

Reply:
removed.